ehrapy/io/_read.py

```diff
@@ -359,6 +351,98 @@ def _do_read_h5ad(filename: Path | Iterator[str]) -> AnnData:
+
+    Args:
+        dataset_path: Path to one or multiple FHIR files.
+        format: The file format of the FHIR data. One of 'json' or 'ndjson'. Defaults to 'json'.
+        columns_obs_only: These columns will be added to obs only and not X.
+        columns_x_only: These columns will be added to X only and all remaining columns to obs. Note that datetime columns will always be added to .obs though.
+        return_df: Whether to return one or several Pandas DataFrames.
+        cache: Whether to write to cache when reading or not. Defaults to False.
+        download_dataset_name: Name of the file or directory in case the dataset is downloaded
+        index_column: The index column for the generated object. Usually the patient or visit ID.
+        backup_url: URL to download the data file(s) from if not yet existing.
+
+    Returns:
+        A Pandas DataFrame or AnnData object of the read in FHIR file(s).
+    """
+    _check_columns_only_params(columns_obs_only, columns_x_only)
+    file_path: Path = Path(dataset_path)
+    if not file_path.exists():
+        file_path = _get_non_existing_files(file_path, download_dataset_name, backup_url)
+
+    adata = _read_fhir(
+        filename=str(file_path.resolve()),
+        format=format,
+        index_column=index_column,
+        columns_obs_only=columns_obs_only,
+        columns_x_only=columns_x_only,
+        return_df=return_df,
+        cache=cache,
+    )
+    return adata
+
+
+def _read_fhir(
+    filename: str,
+    format: Literal["json", "ndjson"],
+    index_column: dict[str, str | int] | str | int | None,
+    columns_obs_only: list[str] | None,
+    columns_x_only: list[str] | None,
+    return_df: bool = False,
+    cache: bool = False,
+) -> AnnData | dict[str, AnnData] | MuData:
+    """Internal interface of the read_csv method."""
+    if cache and return_df:
+        raise CachingNotSupported("Caching is currently not supported for MuData or Pandas DataFrame objects.")
+    if return_df and (columns_x_only or columns_obs_only):
+        raise Warning(
+            "Parameters columns_x_only and columns_obs_only are not supported when returning Pandas DataFrames."
+        )
+    path_cache = settings.cachedir / filename
+    if cache and (path_cache.is_dir() or path_cache.is_file()):
+        raise CacheExistsException(
+            f"{path_cache} already exists. Use the read_h5ad function instead to read from cache!"
+        )
+    if format == "json":
+        df = fp.process(filename)
+    elif format == "ndjson":
+        df = fp.ndjson(filename)
+    else:
+        raise ValueError("Only folders containing json and ndjson in FHIR format are supported.")
+
+    df, columns_obs_only = _prepare_dataframe(df, columns_obs_only, columns_x_only, cache)
+    if index_column:
+        df.set_index(index_column)
+
+    if return_df:
+        return df
+    else:
+        adata = df_to_anndata(df, columns_obs_only)
```

Comment:
don't we need to use "columns_x_only" here as well?

Reply:
No. The function can't take this parameter (yet). Any reason why?